qiskit_nature/second_q/operators/polynomial_tensor.py

```diff
@@ -167,11 +347,13 @@ def _add(self, other: PolynomialTensor, qargs=None) -> PolynomialTensor:
 
         return PolynomialTensor(sum_dict, validate=False)
 
+    # pylint: disable=too-many-return-statements
     def __eq__(self, other: object) -> bool:
-        """Check equality of first PolynomialTensor with other
+        """Check equality of ``PolynomialTensor`` instances.
```

Comment:
Just a comment - here equality is based on values of the elements of the matrices - the representation (sparse or dense) does not matter,